pilot/pkg/model/service.go

```diff
@@ -58,7 +67,7 @@ import (
 type Service struct {
 	// Attributes contains additional attributes associated with the service
 	// used mostly by RBAC for policy enforcement purposes.
-	Attributes ServiceAttributes
+	Attributes *ServiceAttributes
```

Comment:
I think making this a metric degrades this a bit. I assume this is to make the mutex work. Would it be simpler to instead make L54 `ClusterVIPs *cluster.AddressMap`?

Reply:
I assume you meant "pointer"? In what way are you thinking it degrades this? Are you thinking from a perf perspective?

Reply:
(likely very) minor perf hit and we either need to make sure all Services have this set or all callers have nil checks. Its nice to just have a plain struct that we never need to worry about these

Reply:
Yeah I struggled with this too ... it is nice to just have a plain struct. The reason I went down this path is that the existence of creation of the ClusterVIPs seems to need protection. If we make the structure dynamically created, we would have to move the mutex up a level which gets ugly. By making it a plain struct within `ServiceAttributes`, we avoid the issue. The down side is that every time we create service, we have to also create `ServiceAttributes`, which is a bit annoying but doable.